src/pip/_internal/commands/show.py

```diff
@@ -140,41 +141,109 @@ def get_requiring_packages(package_name):
+    Build a dictionary with  information from installed distributions found.
     """
-    results_printed = False
+    pkg_infos_list = []
+
     for i, dist in enumerate(distributions):
-        results_printed = True
-        if i > 0:
-            write_output("---")
 
-        write_output("Name: %s", dist.get('name', ''))
-        write_output("Version: %s", dist.get('version', ''))
-        write_output("Summary: %s", dist.get('summary', ''))
-        write_output("Home-page: %s", dist.get('home-page', ''))
-        write_output("Author: %s", dist.get('author', ''))
-        write_output("Author-email: %s", dist.get('author-email', ''))
-        write_output("License: %s", dist.get('license', ''))
-        write_output("Location: %s", dist.get('location', ''))
-        write_output("Requires: %s", ', '.join(dist.get('requires', [])))
-        write_output("Required-by: %s", ', '.join(dist.get('required_by', [])))
+        pkg_info = {}
+
+        pkg_info["Name"] = dist.get('name', '')
+        pkg_info["Version"] = dist.get('version', '')
+        pkg_info["Summary"] = dist.get('summary', '')
+        pkg_info["Home-page"] = dist.get('home-page', '')
+        pkg_info["Author"] = dist.get('author', '')
+        pkg_info["Author-email"] = dist.get('author-email', '')
+        pkg_info["License"] = dist.get('license', '')
+        pkg_info["Location"] = dist.get('location', '')
+        pkg_info["Requires"] = dist.get('requires', [])
+        pkg_info["Required-by"] = dist.get('required_by', [])
 
         if verbose:
-            write_output("Metadata-Version: %s",
-                         dist.get('metadata-version', ''))
-            write_output("Installer: %s", dist.get('installer', ''))
-            write_output("Classifiers:")
+            pkg_info["Metadata-Version"] = dist.get('metadata-version', '')
+            pkg_info["Installer"] = dist.get('installer', '')
+
+            pkg_info["Classifiers"] = []
             for classifier in dist.get('classifiers', []):
-                write_output("  %s", classifier)
-            write_output("Entry-points:")
+                pkg_info["Classifiers"].append(classifier)
+
+            pkg_info["Entry-points"] = []
             for entry in dist.get('entry_points', []):
-                write_output("  %s", entry.strip())
+                pkg_info["Entry-points"].append(entry.strip())
+
         if list_files:
-            write_output("Files:")
-            for line in dist.get('files', []):
-                write_output("  %s", line.strip())
             if "files" not in dist:
+                pkg_info["Files"] = None
+            else:
+                pkg_info["Files"] = []
+                for line in dist.get('files', []):
+                    pkg_info["Files"].append(line.strip())
+
+        pkg_infos_list.append(pkg_info)
+
+    return pkg_infos_list
+
+
+def print_results(distributions, option='default', list_files=False,
```

Comment:
I think this function could go away.
We'll check the `--format` option in `run()` and call the appropriate "print_" function accordingly.

Reply:
Hi @xavfernandez 

Got rid of the function and hardcoded the option for now. Will add the command-line argument for option in a followup  PR